diff --git a/Source/cmake/OptionsJSCOnly.cmake b/Source/cmake/OptionsJSCOnly.cmake
index c5c859495289..bf5f1922e818 100644
--- a/Source/cmake/OptionsJSCOnly.cmake
+++ b/Source/cmake/OptionsJSCOnly.cmake
@@ -113,6 +113,11 @@ else ()
     SET_AND_EXPOSE_TO_BUILD(WTF_DEFAULT_EVENT_LOOP 0)
 endif ()
 
+if (ENABLE_REMOTE_INSPECTOR AND UNIX AND NOT APPLE AND LOWERCASE_EVENT_LOOP_TYPE STREQUAL "generic")
+    SET_AND_EXPOSE_TO_BUILD(USE_INSPECTOR_SOCKET_SERVER ON)
+    SET_AND_EXPOSE_TO_BUILD(USE_UNIX_DOMAIN_SOCKETS ON)
+endif ()
+
 find_package(ICU 70.1 REQUIRED COMPONENTS data i18n uc)
 if (APPLE)
     add_definitions(-DU_DISABLE_RENAMING=1)
